sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -1231,6 +1356,67 @@ await foreach (var partitionEvent in consumer.ReadEventsFromPartitionAsync(parti
                 }
             });
 
+        /// <summary>
+        ///   Applies the processor's <see cref="RetryPolicy" /> to a specified function.
+        /// </summary>
+        ///
+        /// <param name="functionToRetry">The function to which the retry policy should be applied.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <typeparam name="T">The type returned by the function to be executed.</typeparam>
+        ///
+        /// <returns>The value returned by the function to which the retry policy has been applied.</returns>
+        ///
+        private async Task<T> ApplyRetryPolicy<T>(Func<CancellationToken, Task<T>> functionToRetry,
+                                                  CancellationToken cancellationToken)
+        {
+            var failedAttemptCount = 0;
```

Comment:
I mentioned it above, but I'll float the thought here too.... should this be delegated to the `StorageManager` to absolve the processor of the need to manage retries?   Since the consumer already does on delegation, it would seem that the storage manager would be a good candidate for the responsibility, so that the processor always delegates.